bindings/bindings.py

```diff
@@ -0,0 +1,593 @@
+            if "properties" in schema:
+                required = set(schema.get("required", []))
+                members = {
+                    k: Parameter(k, classify_type(path, v), (k in required), "definitions") for k, v in schema["properties"].items()
+                }
+                # TODO create parameters
+                defs[name] = Class(name, members)
+                continue
+            else:
+                # empty responses or empty requests... we don't care.
+                defs[name] = None
+                continue
+        raise ValueError(f"unhandled schema: {schema} @ {path}")
+    return defs
+
+
+def process_paths(swagger_paths: dict, defs: TypeDefs) -> typing.Dict[str, Function]:
+    ops = {}
+    for path, methods in swagger_paths.items():
+        for method, spec in methods.items():
+            name = spec["operationId"]
+            # Figure out response types.
+            responses = {}
+            bad_op = False
+            for code, rspec in spec["responses"].items():
+                if rspec.get("schema", {}).get("title", "").startswith("Stream result"):
+                    # TODO: support streaming endpoints.
+                    bad_op = True
+                    break
+                if rspec["schema"].get("type") == "":
+                    # not a valid response schema, skipping
+                    bad_op = True
+                    break
+                responses[code] = classify_type(f"{name}.responses.{code}", rspec["schema"])
+            if bad_op:
+                continue
+
+            # Figure out parameters.
+            params = {}
+            for pspec in spec.get("parameters", []):
+                where = pspec["in"]
+                serialized_name = None
+                if where == "query": # preserve query parameter names
+                    serialized_name = pname = pspec["name"]
+                pname = pspec["name"].replace(".", "_")
+                required = pspec.get("required", False)
+                if "schema" in pspec:
+                    pschema = pspec["schema"]
+                else:
+                    # swagger has some weird inlining going on here...
+                    inlined = ("type", "format", "items", "properties", "enum")
+                    pschema = {k: pspec[k] for k in inlined if k in pspec}
+                ptype = classify_type(f"{name}.{pname}", pschema)
+                params[pname] = Parameter(pname, ptype, required, where, serialized_name)
+
+            # TODO: Validate before altering the whole path.
+            path = path.replace(".", "_")
+            op = Function(name, method, path, params, responses)
+            ops[name] = op
+    return ops
+
+
+def link_all_refs(defs: TypeDefs) -> None:
+    for ref in Ref.all_refs:
+        ref.linked = True
+        ref.defn = defs[ref.name]
+
+
+def pybindings(swagger: dict) -> str:
+    prefix = """
+import enum
+import math
+import typing as t
```

Comment:
we don't really do such alias anywhere. maybe just use it as `typing`?

Reply:
I only did this since it's repeated so many times in the outputted code. does the file size affect how fast the interpreter goes through it?

Reply:
Not particularly, as far as I know.  You could time it.

Reply:
added a todo to do this before landing.